### ISSUE TITLE
A couple of questions about white balance

### ISSUE THREAD

**Issue by ilia3101** [AcademySoftwareFoundation/rawtoaces#117]
I am considering adding EXR export support to [MLV App](https://github.com/ilia3101/MLV-App/issues/197). It is a raw converter (lightroom type app) for the magic lantern raw (MLV) format. Recently, EXR export has been suggested and I like the idea.

I want to make it possible to export ACES compliant EXR files, so I have a couple of questions about how rawtoaces does it:

My understanding is that rawtoaces does the following: converts raw images the ACES AP0 colour gamut, using either matrices generated from spectral data or the adobe coefficients, then exports to scene linear EXR - is my understanding correct?

And what is the white balance method used? Is it the bradford transform? And does it adapt the specified white point (from the user-specified grey box for example) to be at the ACES white point (somewhere near D60)?

**TLDR: what are the rules about white balance for getting ACES compliant EXR images?**

**Comment by antond-weta** [comment]
Hi there, this is a very old issue, have you found the answers to your questions?

Here is some info, so we can close the issue.

Usually an input transform matrix for a camera is 'white preserving', which means it converts (1,1,1) in the camera space to (1,1,1) in the output space. The matrix expects the input to be white-balanced, i.e. a neutral grey pixel should have equal RGB values. White balancing gets applied prior to the matrix transform, and ideally prior to any other pixel manipulations to prevent channel crosstalk. Here is some relevant info https://community.acescentral.com/t/lms-vs-camera-native-for-white-balance/4175

The white-balancing itself is pretty simple, the RGB channels in camera space get multiplied by the weights which would make the neutral grey pixels have equal RGB values. The weights can either be obtained from the image file metadata; or calculated from the camera spectral sensitivity and the illuminant spectral power distribution, if known.

Please note that camera white-balancing is a different process from chromatic adaptation (Bradford, etc), or gamut conversion (XYZ to ACES).
